### PR TITLE
fix bug_@ConditionalOnProperty prevents creating of bean in-casse Azure key vault secrets name is in camelCase

### PR DESCRIPTION
**Pull request by hui1110**
Fix https://github.com/Azure/azure-sdk-for-java/issues/23815

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultPropertySource.java [hunk 1]
```diff
@@ -35,4 +35,9 @@ public Object getProperty(String name) {
         return operations.getProperty(name);
     }
 
+    @Override
+    public boolean containsProperty(String name) {
+        return (getProperty(name) != null);
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultPropertySource.java:40]: We could remove the outter "()", @chenrujun is this the way to fix the bug? Is the return value of `getPropertyNames()` correct now?

Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultPropertySource.java]: > this the way to fix the bug?

Yes.

> Is the return value of getPropertyNames() correct now?

No. It's still possible that a key can get value by `getProperty()`, but the key does not belong to the value returned by `getPropertyNames()`, it's a problem that first introduced here: https://github.com/microsoft/azure-spring-boot/pull/550/commits/cd12e47a465798d9b8ec2b6202dddd488205c508

**Review by chenrujun — APPROVED** [review]
LGTM.

**Comment by chenrujun** [comment]
@hui1110 
Could you please add this in to change log?
Just like this one: https://github.com/Azure/azure-sdk-for-java/pull/27745

**Pull request merged**